### PR TITLE
`IntegrationTests`: added test to verify `Purchases.simulatesAskToBuyInSandbox`

### PR DESCRIPTION
**Pull request by NachoSoto**
Finishes [sc-7662].

### CODE REVIEW

**Comment by NachoSoto** [comment]
This is very much a WIP, would love some help from @aboedo to get this running locally so I can debug it and finish it :D

**Comment by aboedo** [comment]
@NachoSoto happy to help! Let's chat over slack

**Comment by NachoSoto** [comment]
I'm suddenly in love with integration tests.

File: BackendIntegrationTests/BackendIntegrationTests.swift [hunk 1]
```diff
@@ -48,6 +48,7 @@ class BackendIntegrationSK1Tests: XCTestCase {
         }
 
         testSession = try SKTestSession(configurationFileNamed: Constants.storeKitConfigFileName)
+        testSession.resetToDefaultState()
```
Comment (NachoSoto) [inline BackendIntegrationTests/BackendIntegrationTests.swift:51]: Even though we're creating a new session for every test, after the new test sets `askToBuyEnabled` to `true`, the following tests inherit this, so internally the test session seems to be shared. This ensures all tests start from a clean state.

File: BackendIntegrationTests/BackendIntegrationTests.swift [hunk 2]
```diff
@@ -227,9 +253,9 @@ private extension BackendIntegrationSK1Tests {
     func verifyEntitlementWentThrough() {
         expect(self.purchasesDelegate.customerInfo?.entitlements.all.count) == 1
```
Comment (NachoSoto) [inline BackendIntegrationTests/BackendIntegrationTests.swift:228]: Wait this should be `active` not `all`?

**Review by aboedo — APPROVED** [review]

**Pull request merged**